src/github.com/couchbase/sync_gateway/db/change_cache.go

```diff
@@ -306,6 +306,27 @@ func (c *changeCache) DocChanged(docID string, docJSON []byte) {
 			c.processEntry(change)
 		}
 
+		// If the recent sequence history includes any sequences earlier than the current sequence, and
+		// not already seen by the gateway (more recent than c.nextSequence), add them as empty entries.
+		currentSequence := doc.Sequence
```

Comment:
This is documenting the "what" but not the "why".  Why does adding empty entries instead of the specified sequences solve any problem(s)?